### DISCUSSION TITLE
Can I do  on/off zod scheme validate conditionally?

### DISCUSSION THREAD

**Discussion by walt1992** [react-hook-form/react-hook-form#10501] [Q&A]
Hello, I'm building a form with Zod resolver.

Here is a part of my form:

```
aField: z.record(z.string(), z.string().min(1, PREFERRED_DELIVERY_DATE_MESSAGE.DATE_REQUIRED)),
```
aField is a key-value data structure where the key represents an object ID and the value represents a date for that object ID.

In most cases, when the key exists, the value is required. However, under certain conditions, the value is not required even if the key exists. In such cases, I would like to disable the validation for this value. 
But. It should be enabled again when the disable condition is ended.

How can I do this?

**Comment by Moshyfawn** [comment]
See if Zod's [`.refine`](https://github.com/colinhacks/zod#refine) fits your use-case

**Reply by walt1992** [reply]
Thanks for your suggestion.
But, the condition is come from out of form data. so It can not be controlled from `.refine`.

**Comment by Bibhushan336** [comment]
hello is this solved? @walt1992

**Reply by sethu1012** [reply]
Use passthrough method. https://zod.dev/?id=passthrough
